pos_default_empty_image/static/src/js/widgets.js

```diff
@@ -0,0 +1,42 @@
+odoo.define('pos_default_empty_image.widgets', function (require) {
+    "use strict";
+
+    var screens = require('point_of_sale.screens');
+
+    var core = require('web.core');
+
+    var QWeb = core.qweb;
+
+    //don't try to get an image if we know the product ain't one
+    var ProductListImageWidget = screens.ProductListWidget.include({
+        get_product_image_url: function(product){
+            if (product.has_image)
+                return this._super(product);
+        },
+
+        // Change product display if product has no image;
+        render_product: function(product){
+            if (product.has_image){
+                return this._super(product);
+            }
+            else {
+                var current_pricelist = this._get_active_pricelist();
+                var cache_key = this.calculate_cache_key(product, current_pricelist);
+                var cached = this.product_cache.get_node(cache_key);
+                if(!cached){
+                    var product_html = QWeb.render('ProductNoImage',{
```

Comment:
What I still wonder is, why not avoid this code and override instead the qweb template, adding a simple `t-if` to the image container (or where necessary to get the right format)

Reply:
@chienandalu I see what you mean but I don't think refactoring that way would be simpler as the whole <article...> section has to be rewritten in the template because it's totally different. If I misunderstood your point, could you help please ?

Reply:
Indeed, t-if is not easy to implement, and I think it will make the code bigger.